simulator/src/test/java/com/hedera/block/simulator/grpc/PublishStreamGrpcClientImplTest.java

```diff
@@ -38,13 +40,15 @@ class PublishStreamGrpcClientImplTest {
 
     GrpcConfig grpcConfig;
     BlockStreamConfig blockStreamConfig;
+    AtomicBoolean streamEnabled;
 
     @BeforeEach
     void setUp() throws IOException {
 
         grpcConfig = TestUtils.getTestConfiguration().getConfigData(GrpcConfig.class);
         blockStreamConfig = TestUtils.getTestConfiguration(Map.of("blockStream.blockItemsBatchSize", "2"))
                 .getConfigData(BlockStreamConfig.class);
+        streamEnabled = new AtomicBoolean(true);
```

Comment:
Based on [this comment](https://github.com/hashgraph/hedera-block-node/pull/327#discussion_r1824545774) if you chose to remove the boolean from the constructor of the grpc client and to DI the `PublishStreamObserver`, you can now mock the stream observer and tell it what to do, in that way you will have control over the internal boolean with mockito.